### PR TITLE
feat: Continue Learning button added on topics and courses

### PR DESCRIPTION
**Pull request by nehasacher143**
Task: https://app.asana.com/0/1194051119161697/1200048723927506/f

GIF for Topics:
Peek 2021-04-06 20-21

GIF for Courses:
Peek 2021-04-07 12-07

### CODE REVIEW

File: erpnext/www/lms/topic.py [hunk 1]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
 		if content.get("completed"):
 			completed_content = completed_content + 1
 	total_progress = int((completed_content * 100) / total_content)
-	return total_progress
\ No newline at end of file
+	return total_progress
+
+def get_ongoing_topic(contents):
+	ongoing_content = ""
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
	ongoing_content = None
```

File: erpnext/www/lms/topic.py [hunk 2]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
...
+	ongoing_topics = []
+	ongoing_topics_length = 0
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
```

File: erpnext/www/lms/topic.py [hunk 3]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
...
+	total_content = len(contents)
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
```

File: erpnext/www/lms/topic.py [hunk 4]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
...
+	for content in contents:
+		if not content.get("completed"):
+			ongoing_topics.append(content)
+
+	ongoing_topics_length = len(ongoing_topics)
+	if total_content != ongoing_topics_length:
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
	if len(contents) != len(ongoing_topics):
```

File: erpnext/www/lms/topic.py [hunk 5]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
...
+def get_ongoing_topic(contents):
+	ongoing_content = ""
+	ongoing_topics = []
+	ongoing_topics_length = 0
+	total_content = len(contents)
+	for content in contents:
+		if not content.get("completed"):
+			ongoing_topics.append(content)
+
+	ongoing_topics_length = len(ongoing_topics)
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
```

File: erpnext/www/lms/topic.py [hunk 6]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
...
+	return total_progress
+
+def get_ongoing_topic(contents):
+	ongoing_content = ""
+	ongoing_topics = []
+	ongoing_topics_length = 0
+	total_content = len(contents)
+	for content in contents:
+		if not content.get("completed"):
+			ongoing_topics.append(content)
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
	ongoing_topics = [content for content in contents if not content.get("completed")]
```

File: erpnext/www/lms/topic.py [hunk 7]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
 		if content.get("completed"):
 			completed_content = completed_content + 1
 	total_progress = int((completed_content * 100) / total_content)
-	return total_progress
\ No newline at end of file
+	return total_progress
+
+def get_ongoing_topic(contents):
+	ongoing_content = ""
+	ongoing_topics = []
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]:
```suggestion
```

File: erpnext/www/lms/topic.py [hunk 8]
```diff
@@ -61,4 +61,22 @@ def calculate_contents_progress(contents):
...
+	total_content = len(contents)
+	for content in contents:
+		if not content.get("completed"):
+			ongoing_topics.append(content)
+
+	ongoing_topics_length = len(ongoing_topics)
+	if total_content != ongoing_topics_length:
+		for ongoing_topic in ongoing_topics:
+			ongoing_content = ongoing_topic.get('content')
+			break
```
Comment (hrwX) [inline erpnext/www/lms/topic.py]: You can directly return ongoing_topic[0].get('content') since ongoing_topic is a list of incomplete topics

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/www/lms/topic.py [hunk 9]
```diff
@@ -61,4 +61,13 @@ def calculate_contents_progress(contents):
...
 	total_progress = int((completed_content * 100) / total_content)
-	return total_progress
\ No newline at end of file
+	return total_progress
+
+def get_ongoing_topic(contents):
+	ongoing_content = None
+	ongoing_topics = [content for content in contents if not content.get("completed")]
+	if len(contents) != len(ongoing_topics):
+		ongoing_content = ongoing_topics[0].get('content')
```
Comment (hrwX) [inline erpnext/www/lms/topic.py:70]: Add a check to confirm if there is a zeroth element, else this will break

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/www/lms/topic.py [hunk 10]
```diff
@@ -61,4 +61,13 @@ def calculate_contents_progress(contents):
...
 			completed_content = completed_content + 1
 	total_progress = int((completed_content * 100) / total_content)
-	return total_progress
\ No newline at end of file
+	return total_progress
+
+def get_ongoing_topic(contents):
+	ongoing_content = None
+	ongoing_topics = [content for content in contents if not content.get("completed")]
+	if len(contents) != len(ongoing_topics):
```
Comment (sahil28297) [inline erpnext/www/lms/topic.py]:
```suggestion
	if len(contents) != len(ongoing_topics) and ongoing_topics[0]:
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/www/lms/topic.py [hunk 11]
```diff
@@ -61,4 +61,12 @@ def calculate_contents_progress(contents):
...
 			completed_content = completed_content + 1
 	total_progress = int((completed_content * 100) / total_content)
-	return total_progress
\ No newline at end of file
+	return total_progress
+
+def get_ongoing_topic(contents):
+	ongoing_content = None
+	ongoing_topics = [content for content in contents if not content.get("completed")]
+	if len(contents) != len(ongoing_topics) and ongoing_topics[0]:
```
Comment (sahil28297) [inline erpnext/www/lms/topic.py]:
```suggestion
	if len(contents) != len(ongoing_topics) and len(ongoing_topics) > 0:
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**